bolts/data/datasets/audio/ecoacoustics.py

```diff
@@ -128,14 +171,40 @@ def _download_files(self) -> None:
         self.root.mkdir(parents=True, exist_ok=True)
         self.base_dir.mkdir(parents=True, exist_ok=True)
 
-        urls = [
-            "https://zenodo.org/record/1255218/files/AvianID_AcousticIndices.zip",
-            "https://zenodo.org/record/1255218/files/EC_BIRD.zip",
-            "https://zenodo.org/record/1255218/files/UK_BIRD.zip",
-        ]
+        for finfo in self._INDICES_URL_MD5_LIST:
+            if not self._check_integrity(finfo.filename, finfo.md5):
+                download_and_extract_archive(
+                    url=finfo.url, download_root=str(self.base_dir), md5=finfo.md5
+                )
+
+        for finfo in self._URL_MD5_LIST:
+            if not self._check_integrity(finfo.filename, finfo.md5):
+                self.download_and_extract_archive_jar(url=finfo.url, md5=finfo.md5)
+
+    def download_and_extract_archive_jar(
+        self,
+        url: str,
+        md5: str,
+        filename: str | None = None,
+        remove_finished: bool = False,
+    ) -> str | None:
+        if not filename:
+            filename = os.path.basename(url)
+
+        download_url(url, str(self.base_dir), filename, md5)
+
+        archive = self.base_dir / filename
+        self.log(f"Extracting {archive}")
+
+        suffix, archive_type, compression = _detect_file_type(str(archive))
+        if not archive_type:
+            return _decompress(
+                str(archive),
+                os.path.join(self.base_dir, filename.replace(suffix, "")),
+                remove_finished=remove_finished,
+            )
 
-        for url in urls:
-            download_and_extract_archive(url, str(self.base_dir))
+        subprocess.run(f"jar -xvf {archive}", shell=True, check=True, cwd=self.base_dir)
```

Comment:
Can we have some kind of error-handling here (e.g. informing the user that jar needs to be installed)?